File: backend/src/main/java/com/runway/course/dto/NearbyCourseItem.java
package com.runway.course.dto;

import lombok.Builder;
import lombok.Getter;

import java.util.List;
import java.util.UUID;

@Getter
@Builder(toBuilder = true)
public class NearbyCourseItem {

    private UUID courseId;
    private String name;
    private String description;
    private Double distanceMeters;
    private Double distanceFromMeMeters;
    private Boolean isLoop;
    private Integer attemptCount;
    private Integer completionCount;
    private GeoPoint startPoint;
    private Double avgRating;
    private Long ratingCount;
    private List<GeoPoint> routePoints;
}

File: android/app/src/main/java/com/runway/android/data/course/model/NearbyCourseItem.kt
package com.runway.android.data.course.model

data class NearbyCourseItem(
    val courseId: String,
    val name: String,
    val description: String?,
    val status: String,
    val distanceMeters: Double,
    val isLoop: Boolean,
    val startPoint: GeoPoint,
    val endPoint: GeoPoint?,
    val attemptCount: Int,
    val completionCount: Int,
    val distanceFromMeMeters: Double,
    val createdAt: String,
    val avgRating: Double?,
    val ratingCount: Long?,
)
